pkg/fleet-manager/application/rollout_helper.go

```diff
@@ -0,0 +1,444 @@
+		// Set up labels to make sure it's a resource created by kurator
+		canaryInCluster.SetLabels(label)
+
+		if getErr != nil {
+			if apierrors.IsNotFound(getErr) {
+				if err := newClient.Create(ctx, canaryInCluster); err != nil {
+					return ctrl.Result{}, fmt.Errorf("failed to create rolloutPolicy: %v", err)
+				}
+			}
+			return ctrl.Result{}, errors.Wrapf(getErr, "failed to get canary %s in %s", namespaceName.Name, namespaceName.Namespace)
+		}
+		if err := newClient.Update(ctx, canaryInCluster); err != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to update rolloutPolicy: %v", err)
+		}
+
+		log.Info("sync rolloutPolicy for cluster successful")
+	}
+	return ctrl.Result{}, nil
+}
+
+func namespaceSidecarInject(ctx context.Context, cluster *fleetmanager.FleetCluster, namespacedName types.NamespacedName) error {
+	ns := &corev1.Namespace{}
+	client := cluster.Client.CtrlRuntimeClient()
+	namespacedName.Name = namespacedName.Namespace
+	if err := client.Get(ctx, namespacedName, ns); err != nil {
+		// if no found, create a namespace
+		if apierrors.IsNotFound(err) {
+			ns.SetName(namespacedName.Namespace)
+			ns.SetLabels(map[string]string{
+				"istio-injection": "enabled",
+			})
+			if createErr := client.Create(ctx, ns); createErr != nil {
+				return errors.Wrapf(createErr, "failed to create namespace %s", namespacedName.Namespace)
+			}
+		}
+		// if found namespace, set labels and update.
+		ns.SetLabels(map[string]string{
+			"istio-injection": "enabled",
+		})
+		if updateErr := client.Update(ctx, ns); updateErr != nil {
+			return errors.Wrapf(updateErr, "failed to update namespace %s", namespacedName.Namespace)
+		}
+	}
+	return nil
+}
+
+func installPrivateTestloader(ctx context.Context,
+	namespacedName types.NamespacedName,
+	fleetCluster fleetmanager.FleetCluster,
+	labels map[string]string,
+) (ctrl.Result, error) {
+	log := ctrl.LoggerFrom(ctx)
+	clusterClient := fleetCluster.Client.CtrlRuntimeClient()
+
+	// Creating a private testload deployment from a configuration file
+	filepath := manifests.BuiltinOrDir("")
+	deployname := "plugins/testloader-deploy.yaml"
+	deploy, err1 := generateDeployConfig(filepath, deployname, namespacedName.Name, namespacedName.Namespace)
+	if err1 != nil {
+		return ctrl.Result{}, fmt.Errorf("failed get testloader deployment configuration: %v", err1)
+	}
+	// Set up labels to make sure it's a resource created by kurator.
+	deploy.SetLabels(labels)
+
+	if err := clusterClient.Create(ctx, deploy); err != nil {
+		if apierrors.IsAlreadyExists(err) {
+			if updateErr := clusterClient.Update(ctx, deploy); updateErr != nil {
+				return ctrl.Result{}, errors.Wrapf(err, "failed to update private testloader deployment")
+			}
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, errors.Wrapf(err, "failed to create private testloader deployment")
+	}
+
+	// Creating a private testload service from a configuration file
+	svcName := "plugins/testloader-svc.yaml"
```

Comment:
nit: Consider using constants for file pathsenhance maintainability and readability.